instrumentation/rocketmq/rocketmq-client/rocketmq-client-5.0/testing/src/main/java/io/opentelemetry/instrumentation/rocketmqclient/v5_0/RocketMqProxyContainer.java

```diff
@@ -6,34 +6,41 @@
 package io.opentelemetry.instrumentation.rocketmqclient.v5_0;
 
 import io.opentelemetry.instrumentation.test.utils.PortUtils;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import org.testcontainers.DockerClientFactory;
 import org.testcontainers.containers.FixedHostPortGenericContainer;
 import org.testcontainers.containers.GenericContainer;
 
-// TODO - Tests using this container will currently fail if TESTCONTAINERS_HOST_OVERRIDE is set to
-// anything other than "localhost".
 public class RocketMqProxyContainer {
   // TODO(aaron-ai): replace it by the official image.
-  private static final String IMAGE_NAME = "aaronai/rocketmq-proxy-it:v1.0.1";
+  private static final String IMAGE_NAME = "aaronai/rocketmq-proxy-it:v1.0.2";
 
   private final GenericContainer<?> container;
   final String endpoints;
 
   // We still need this container type to do fixed-port-mapping.
   @SuppressWarnings({"resource", "deprecation", "rawtypes"})
-  RocketMqProxyContainer() {
+  RocketMqProxyContainer() throws UnknownHostException {
```

Comment:
I think it would be better to catch the `UnknownHostException` inside this method around the `getByName` call and wrap it inside `IllegalStateException` so callers wouldn't have to deal with this.

Reply:
That's what I originally did, this change came about due to this conversation on the original PR: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11106#discussion_r1576904546

I'll change it back, though, because it seems strange to me as well.